client/client.go

```diff
@@ -0,0 +1,182 @@
+// Package client provides a Certificate Transparency (CT) Log Client that
+// prioritizes preserving and returning as much information about each http
+// request that is made, and detailed descriptions of any errors that occur
+// along the way.
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+)
+
+// LogClient is a client for a sepcific CT Log.
+type LogClient struct {
+	url        string
+	HttpClient *http.Client
+}
+
+// New creates a new LogClient for monitoring the CT Log served at logURL.
+func New(logURL string, hc *http.Client) *LogClient {
+	return &LogClient{url: strings.TrimRight(logURL, "/"), HttpClient: hc}
```

Comment:
Why is the `TrimRight` done here, it doesn't really seem to be necessary (from my reading of `BuildURL`)?

Reply:
I have dropped the strings.TrimRight (both here and in sthgetter) after all of our experimenting.  However I haven't got rid of the New function yet as I'm going to keep the fields internal to the LogClient unexported until something needs them, as really they shouldn't be changed after the LC has been created.

But who knows, this may change in the future, and if it does, I'll drop New then :)